e2e-tests/desktop/tests/credentials.spec.js

```diff
@@ -0,0 +1,304 @@
+      });
+
+    // Create a tcp target with vault brokered credential Libraries
+    targetWithBrokeredVaultCredentials = await boundaryHttp.createTarget(
+      request,
+      {
+        scopeId: project.id,
+        type: 'tcp',
+        port: targetPort,
+      },
+    );
+    targetWithBrokeredVaultCredentials = await boundaryHttp.addHostSource(
+      request,
+      {
+        target: targetWithBrokeredVaultCredentials,
+        hostSourceIds: [hostSet.id],
+      },
+    );
+    targetWithBrokeredVaultCredentials =
+      await boundaryHttp.addBrokeredCredentials(request, {
+        target: targetWithBrokeredVaultCredentials,
+        credentialIds: [vaultGenericCredentialLibrary.id],
+      });
+
+    // Create a tcp target with static brokered credentials
+    targetWithBrokeredStaticCredentials = await boundaryHttp.createTarget(
+      request,
+      {
+        scopeId: project.id,
+        type: 'tcp',
+        port: targetPort,
+      },
+    );
+    targetWithBrokeredStaticCredentials = await boundaryHttp.addHostSource(
+      request,
+      {
+        target: targetWithBrokeredStaticCredentials,
+        hostSourceIds: [hostSet.id],
+      },
+    );
+    targetWithBrokeredStaticCredentials =
+      await boundaryHttp.addBrokeredCredentials(request, {
+        target: targetWithBrokeredStaticCredentials,
+        credentialIds: [sshPrivateKeyCredential.id, jsonCredential.id],
+      });
+  },
+);
+
+test.afterEach(async ({ request }) => {
+  if (org) {
+    await boundaryHttp.deleteOrg(request, org.id);
+  }
+});
+
+test.describe('Credential Panel tests', async () => {
+  test('Display Vault brokered Credentials and handle special characters', async ({
+    authedPage,
+  }) => {
+    await authedPage
+      .getByRole('link', { name: targetWithBrokeredVaultCredentials.name })
+      .click();
+    await authedPage.getByRole('button', { name: 'Connect' }).click();
+    await expect(
+      authedPage.getByRole('heading', { name: 'Sessions' }),
+    ).toBeVisible();
+
+    await expect(
+      authedPage.getByText(vaultGenericCredentialLibrary.name),
+    ).toBeVisible();
+    await expect(authedPage.getByText('password')).toBeVisible();
+
+    await authedPage
+      .getByRole('listitem')
+      .filter({ hasText: 'password' })
+      .getByLabel('Toggle secret visibility')
+      .click();
+    await expect(
+      authedPage
+        .getByRole('listitem')
+        .filter({ hasText: 'password' })
+        .locator('pre'),
+    ).toHaveText(String.raw`pass\word`);
+
+    // End session as active sessions will show a popup when trying to close the DC
+    await authedPage.getByRole('button', { name: 'End Session' }).click();
+    await expect(authedPage.getByText('Canceled successfully.')).toBeVisible();
+    await expect(
+      authedPage.getByRole('heading', { name: 'Targets' }),
+    ).toBeVisible();
+  });
+
+  test('Display JSON static credential in a key value format and handle special characters', async ({
```

Comment:
How come we don't end session in this test? I also get the popup at the end of this test

Reply:
oh it should be added, I also received a popup at the end of the test and missed to add it here, thank you for testing!